packages/firebase_auth/firebase_auth_web/lib/src/utils/web_utils.dart

```diff
@@ -204,10 +205,18 @@ AuthCredential? convertWebOAuthCredential(
 
   return OAuthProvider(oAuthCredential.providerId).credential(
     accessToken: oAuthCredential.accessToken,
-    idToken: oAuthCredential.idToken,
+    idToken: oAuthCredential.idToken as String,
   );
 }
 
+@JS()
```

Comment:
Rather than [this](https://github.com/FirebaseExtended/flutterfire/pull/6597/files#diff-a522fde29aeb319e6149a657c49c90dcce5c910e0e1166a1b37a0f00b4360885R212-R219), create a [OAuthCredentialOptions](https://firebase.google.com/docs/reference/js/firebase.auth.OAuthCredentialOptions) class. Use this [firebase JS SDK class](https://github.com/FirebaseExtended/flutterfire/blob/ec314b80d5badf29013c644a451c708de57e6f67/packages/firebase_auth/firebase_auth_web/lib/src/interop/auth_interop.dart#L209-L221) for implementation inspiration.

Reply:
I moved and renamed the class into auth_interop.dart 
https://github.com/FirebaseExtended/flutterfire/pull/6597/commits/26960339350e6883bd7cfbdaf00ecc8e19d42b8a#diff-0bc8e1dd3fd419247b4814219db78766ee030355372fa17bb3091c88278f4fa4R228